packages/flutter/test/material/navigation_bar_test.dart

```diff
@@ -0,0 +1,429 @@
+          destinations: const <Widget>[
+            NavigationDestination(
+              icon: Icon(Icons.ac_unit),
+              label: 'AC',
+            ),
+            NavigationDestination(
+              icon: Icon(Icons.access_alarm),
+              label: 'Alarm',
+            ),
+          ],
+        ),
+      );
+    }
+
+    await tester.pumpWidget(_widget(selectedIndex: 0));
+
+    expect(
+      tester.getSemantics(find.text('AC')),
+      matchesSemantics(
+        label: 'AC\nTab 1 of 2',
+        textDirection: TextDirection.ltr,
+        isFocusable: true,
+        isSelected: true,
+        hasTapAction: true,
+      ),
+    );
+    expect(
+      tester.getSemantics(find.text('Alarm')),
+      matchesSemantics(
+        label: 'Alarm\nTab 2 of 2',
+        textDirection: TextDirection.ltr,
+        isFocusable: true,
+        isSelected: false,
+        hasTapAction: true,
+      ),
+    );
+
+    await tester.pumpWidget(_widget(selectedIndex: 1));
+
+    expect(
+      tester.getSemantics(find.text('AC')),
+      matchesSemantics(
+        label: 'AC\nTab 1 of 2',
+        textDirection: TextDirection.ltr,
+        isFocusable: true,
+        isSelected: false,
+        hasTapAction: true,
+      ),
+    );
+    expect(
+      tester.getSemantics(find.text('Alarm')),
+      matchesSemantics(
+        label: 'Alarm\nTab 2 of 2',
+        textDirection: TextDirection.ltr,
+        isFocusable: true,
+        isSelected: true,
+        hasTapAction: true,
+      ),
+    );
+  });
+
+  testWidgets('Navigation bar does not grow with text scale factor', (WidgetTester tester) async {
+    final Key painterKey = UniqueKey();
+    const int _animationMilliseconds = 800;
+
+    Widget _widget({double textScaleFactor = 1}) {
+      return _buildWidget(
+        painterKey,
+        MediaQuery(
+          data: MediaQueryData(textScaleFactor: textScaleFactor),
+          child: NavigationBar(
+            animationDuration: const Duration(milliseconds: _animationMilliseconds),
+            destinations: const <NavigationDestination>[
+              NavigationDestination(
+                icon: Icon(Icons.ac_unit),
+                label: 'AC',
+              ),
+              NavigationDestination(
+                icon: Icon(Icons.access_alarm),
+                label: 'Alarm',
+              ),
+            ],
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(_widget());
+    final double initialHeight =
+        tester.renderObject<RenderBox>(find.byType(NavigationBar)).size.height;
+
+    await tester.pumpWidget(_widget(textScaleFactor: 2));
+    final double newHeight =
+        tester.renderObject<RenderBox>(find.byType(NavigationBar)).size.height;
+
+    expect(newHeight, equals(initialHeight));
+  });
+}
+
+Widget _buildWidget(Key key, Widget child) {
+  return MaterialApp(
+    theme: ThemeData.light(),
+    home: Scaffold(
+      bottomNavigationBar: Center(
+        child: RepaintBoundary(
```

Comment:
Why create a RepaintBoundary (if there's a good reason, please add a comment).

Reply:
Not needed, removed